internal/fnruntime/nodejs.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fnruntime
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	fnresult "github.com/GoogleContainerTools/kpt/pkg/api/fnresult/v1"
+)
+
+const (
+	WasmPathEnv = "KPT_FN_WASM_PATH"
+)
+
+type WasmNodejsFn struct {
+	NodeJsRunner *ExecFn
+	loader       WasmLoader
+}
+
+func NewNodejsFn(loader WasmLoader) (*WasmNodejsFn, error) {
+	cacheDir := filepath.Join(os.TempDir(), "kpt-wasm-fn")
+	tempDir, err := os.MkdirTemp(cacheDir, "nodejs-")
+	if err != nil {
+		return nil, fmt.Errorf("unable to create temp dir: %w", err)
+	}
+	jsPath := filepath.Join(tempDir, "kpt-fn-wasm-glue-runner.js")
+	if err = os.WriteFile(jsPath, []byte(golangWasmJSCode+glueCode), 0666); err != nil {
```

Comment:
Nit: I think 0644 or 600, because we don't want other people to write to the file.